### PR TITLE
fix(connector): fixes MGDCTRS-659, added endpoints for getting deployments

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Added admin api endpoints to get deployments

## Verification Steps
Included integration tests must pass. 

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
~~- [ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
~~- [ ] Verified independently by reviewer~~
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by dhirajsb** [comment]
@lburgazzoli please review the updated admin API.

File: internal/connector/internal/api/admin/private/api/openapi.yaml [hunk 1]
```diff
@@ -266,14 +266,14 @@ paths:
       summary: Get a list of available connector namespaces in cluster
       tags:
       - Connector Clusters Admin
-  /api/connector_mgmt/v1/admin/kafka_connector_namespaces/${namespace_id}/connectors:
+  /api/connector_mgmt/v1/admin/kafka_connector_clusters/${connector_cluster_id}/connectors:
```
Comment (valdar) [inline internal/connector/internal/api/admin/private/api/openapi.yaml:269]: That $ sign there feels odd, not sure it should be there?

File: internal/connector/internal/api/admin/private/api/openapi.yaml [hunk 2]
```diff
@@ -396,36 +396,74 @@ paths:
...
       security:
       - Bearer: []
-      summary: Get a list of available connectors in a namespace in a cluster
+      summary: Get a list of available connectors in a cluster
       tags:
       - Connector Clusters Admin
-  /api/connector_mgmt/v1/admin/kafka_connectors/${connector_id}:
-    delete:
-      operationId: deleteConnector
+  /api/connector_mgmt/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments:
```
Comment (valdar) [inline internal/connector/internal/api/admin/private/api/openapi.yaml:402]: Same

File: openapi/connector_mgmt-private-admin.yaml [hunk 3]
```diff
@@ -115,13 +115,13 @@ paths:
       operationId: getClusterNamespaces
       summary: Get a list of available connector namespaces in cluster
 
-  /api/connector_mgmt/v1/admin/kafka_connector_namespaces/${namespace_id}/connectors:
+  /api/connector_mgmt/v1/admin/kafka_connector_clusters/${connector_cluster_id}/connectors:
```
Comment (valdar) [inline openapi/connector_mgmt-private-admin.yaml:118]: Same

File: openapi/connector_mgmt-private-admin.yaml [hunk 4]
```diff
@@ -166,28 +166,87 @@ paths:
           description: Unexpected error occurred
       security:
         - Bearer: [ ]
-      operationId: getNamespaceConnectors
-      summary: Get a list of available connectors in a namespace in a cluster
+      operationId: getClusterConnectors
+      summary: Get a list of available connectors in a cluster
 
-  /api/connector_mgmt/v1/admin/kafka_connectors/${connector_id}:
+  /api/connector_mgmt/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments:
```
Comment (valdar) [inline openapi/connector_mgmt-private-admin.yaml:172]: Same

File: internal/kafka/internal/clusters/standalone_provider_test.go [hunk 5]
```diff
@@ -1052,7 +1052,7 @@ func TestStandaloneProvider_InstallStrimzi(t *testing.T) {
 		name    string
 		fields  fields
 		args    args
-		want	bool
+		want    bool
```
Comment (valdar) [inline internal/kafka/internal/clusters/standalone_provider_test.go:1055]: Are these necessary? On paper touching this file means an approval is needed by kafka team...

**Review by valdar — CHANGES_REQUESTED** [review]

File: internal/kafka/internal/clusters/standalone_provider_test.go (hunk 5 above)
Comment (dhirajsb) [reply internal/kafka/internal/clusters/standalone_provider_test.go:1055]: must have come from `make lint`, I'll undo that change.

File: internal/connector/internal/api/admin/private/api/openapi.yaml (hunk 1 above)
Comment (dhirajsb) [reply internal/connector/internal/api/admin/private/api/openapi.yaml:269]: yeah, it's a typo, will remove. Good catch. It's interesting that openapi generator and validation doesn't complain. I guess it expects a `$` sign in the url in this case??

File: internal/kafka/internal/clusters/standalone_provider_test.go (hunk 5 above)
Comment (dhirajsb) [reply internal/kafka/internal/clusters/standalone_provider_test.go]: fixed

File: internal/connector/internal/api/admin/private/api/openapi.yaml (hunk 1 above)
Comment (valdar) [reply internal/connector/internal/api/admin/private/api/openapi.yaml]: yep

**Review by valdar — APPROVED** [review]

File: internal/connector/internal/api/admin/private/api/openapi.yaml (hunk 1 above)
Comment (lburgazzoli) [reply internal/connector/internal/api/admin/private/api/openapi.yaml]: I've opened an issue for that https://issues.redhat.com/browse/MGDCTRS-836
The opening generator does not complain as it interprets the '$' as is so it's just a char (but fails the invocation of endpoints as the IDs are then not found)

**Review by lburgazzoli — APPROVED** [review]

**Pull request merged**